server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/api/routes/JMAPApiRoutes.scala

```diff
@@ -0,0 +1,33 @@
+package org.apache.james.jmap.rfc.api.routes
+
+import org.apache.james.jmap.rfc.api.method.EchoMethod
+import org.apache.james.jmap.rfc.model.{RequestObject, ResponseObject}
+import org.slf4j.{Logger, LoggerFactory}
+import reactor.core.publisher.Mono
+
+object JMAPApiRoutes {
+  private val LOGGER = LoggerFactory.getLogger(JMAPApiRoutes.getClass)
+  private val ECHO_METHOD = new EchoMethod();
+}
+
+class JMAPApiRoutes {
+  val logger: Logger = JMAPApiRoutes.LOGGER
+  val echoMethod: EchoMethod = JMAPApiRoutes.ECHO_METHOD
+
+  def post(httpRequest: HttpServerRequest, httpServerResponse: HttpServerResponse): Mono[Void] = {
+    httpRequest
+      .receive()
+      .asInputStream()
+      .flatMap(inputStream => {
+        echoMethod.toRequestObject(inputStream)
+      })
+      .flatMap((requestObject: RequestObject) => {
+        echoMethod.toResponseObject(requestObject)
```

Comment:
We should have a list of methods, and rely on method name of the invocation object to select the right method to execute from the list

(even if the list contains a single element.)

Reply:
Similar to draft version right?

Reply:
Yes but here we can do pattern matching:

```
methodArguments match {
  case EchoArguments => echoMethod.process(methodArguments)
}
```